deepspeed/runtime/utils.py

```diff
@@ -1065,3 +1065,38 @@ def to_tensor(v):
             total_norm = -1
 
     return total_norm
+
+
+def offload_adam_states(optimizer, device, pin_memory: bool = False, non_blocking: bool = False):
+    """Move optimizer states to device. Note that this assumes the state structure of DeepSpeed Adam."""
+
+    def move_key(state, key):
+        if pin_memory:
+            pin_mem_key = f"{key}_pin_memory"
```

Comment:
Since we never free the offload buffer, I wonder if we need this much difference between pin and non-pin options. For example, would something like this work
```python
  offload_buffer = torch.empty_like(state[key], device=device)
  if pin_memory:
      state['offload_buffer'] = get_accelerator().pin_memory(offload_buffer)
  else:
    state['offload_buffer'] = offload_buffer
 state['offload_buffer'].copy_(state[key,],  ...)
 state[key].data = state['offload_buffer']
```

Reply:
Simplified the logic as suggested